go-controller/pkg/util/net.go

```diff
@@ -389,3 +390,32 @@ func RouteEqual(l, r *netlink.Route) bool {
 		l.Congctl == r.Congctl &&
 		l.FastOpenNoCookie == r.FastOpenNoCookie
 }
+
+// SubnetBroadcastIP returns the IP network's broadcast IP.
+func SubnetBroadcastIP(ipnet net.IPNet) net.IP {
+	// For IPv4.
+	if ipnet.IP.To4() != nil {
+		// ip address in uint32
+		ipBits := binary.BigEndian.Uint32(ipnet.IP.To4())
+		// mask will give us all fixed bits of the subnet
+		maskBits := binary.BigEndian.Uint32(ipnet.Mask)
+		// inverted mask will give us all moving bits of the subnet
+		invertedMaskBits := maskBits ^ 0xffffffff // xor the mask
+		// network ip
+		networkIPBits := ipBits & maskBits
+		// broadcastIP = networkIP added to the inverted mask
+		broadcastIPBits := networkIPBits | invertedMaskBits
+		broadcastIP := make(net.IP, 4)
+		binary.BigEndian.PutUint32(broadcastIP, broadcastIPBits)
+		return broadcastIP
```

Comment:
nit: could we move this to an helper, and invoke it ? not a must

Reply:
I've simplified the function significantly and modfied the logic to only skip the broadcast IP for v4 as it is not a thing in IPv6.
Hope that helps.